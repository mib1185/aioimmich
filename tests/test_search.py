"""Tests for aioimmich."""

from __future__ import annotations

from syrupy.assertion import SnapshotAssertion


async def test_get_all_assets(
    mock_immich_with_data, mock_aioresponse, snapshot: SnapshotAssertion
):
    """Test async_get_all."""
    api = await mock_immich_with_data()
    assets = await api.search.async_get_all()
    mock_aioresponse.assert_called_with(
        "https://localhost:2283/api/search/metadata",
        "post",
        headers={"Accept": "application/json", "x-api-key": "abcdef1234567890"},
        params=None,
        json={"size": 100, "page": 1},
    )
    assert len(assets) == 4
    assert assets == snapshot


async def test_get_all_by_tag_ids(
    mock_immich_with_data, mock_aioresponse, snapshot: SnapshotAssertion
):
    """Test async_get_all_by_tag_ids."""
    api = await mock_immich_with_data()
    assets = await api.search.async_get_all_by_tag_ids(
        ["14ce3af3-67be-41c6-b77c-b25abddaf546"]
    )
    mock_aioresponse.assert_called_with(
        "https://localhost:2283/api/search/metadata",
        "post",
        headers={"Accept": "application/json", "x-api-key": "abcdef1234567890"},
        params=None,
        json={
            "size": 100,
            "tagIds": ["14ce3af3-67be-41c6-b77c-b25abddaf546"],
            "page": 1,
        },
    )
    assert len(assets) == 4
    assert assets == snapshot


async def test_get_all_by_person_ids(
    mock_immich_with_data, mock_aioresponse, snapshot: SnapshotAssertion
):
    """Test async_get_all_by_person_ids."""
    api = await mock_immich_with_data()
    assets = await api.search.async_get_all_by_person_ids(
        ["14ce3af3-67be-41c6-b77c-b25abddaf546"]
    )
    mock_aioresponse.assert_called_with(
        "https://localhost:2283/api/search/metadata",
        "post",
        headers={"Accept": "application/json", "x-api-key": "abcdef1234567890"},
        params=None,
        json={
            "size": 100,
            "personIds": ["14ce3af3-67be-41c6-b77c-b25abddaf546"],
            "page": 1,
        },
    )
    assert len(assets) == 4
    assert assets == snapshot
