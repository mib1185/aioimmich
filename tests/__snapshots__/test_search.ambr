# serializer version: 1
# name: test_get_all_assets
  list([
    ImmichAsset(asset_id='2e94c203-50aa-4ad2-8e29-56dd74e0eff4', asset_type=<AssetType.IMAGE: 'IMAGE'>, checksum='HJm7TVOP80S+eiYZnAhWyRaB/Yc=', device_asset_id='web-20230131_182037.jpg-1675185639000', device_id='WEB', duration='0:00:00.00000', file_created_at=datetime.datetime(2023, 1, 31, 17, 20, 37, 85000, tzinfo=datetime.timezone.utc), file_modified_at=datetime.datetime(2023, 1, 31, 17, 20, 39, tzinfo=datetime.timezone.utc), has_metadata=True, is_archived=False, is_favorite=False, is_offline=False, is_trashed=False, local_datetime=datetime.datetime(2023, 1, 31, 18, 20, 37, 85000, tzinfo=datetime.timezone.utc), original_file_name='20230131_182037.jpg', original_path='upload/upload/e7ef5713-9dab-4bd4-b899-715b0ca4379e/b4/b8/b4b8ef00-8a6d-4056-91ff-7f86dc66e427.jpg', owner_id='e7ef5713-9dab-4bd4-b899-715b0ca4379e', people=[], resized=True, thumbhash='1igGFALX8mVGdHc5aChJf5nxNg==', updated_at=datetime.datetime(2025, 5, 11, 10, 13, 49, 590401, tzinfo=datetime.timezone.utc), duplicate_id=None, exif_info=ExifInfo(city=None, country=None, date_time_original=None, description=None, exif_image_height=None, exif_image_width=None, exposure_time=None, file_size_in_byte=None, f_number=None, focal_length=None, iso=None, latitude=None, lens_model=None, longitude=None, make=None, model=None, modify_date=None, orientation=None, projection_type=None, rating=None, state=None, time_zone=None), library_id=None, live_photo_video_id=None, original_mime_type='image/jpeg', visibility=None),
    ImmichAsset(asset_id='de104b75-6c7f-46a2-83ce-12a5b1c3a3b4', asset_type=<AssetType.IMAGE: 'IMAGE'>, checksum='aNCzp51jAuT3rXMv1icKLGMTC6M=', device_asset_id='web-20221222_205943.jpg-1671739185000', device_id='WEB', duration='0:00:00.00000', file_created_at=datetime.datetime(2022, 12, 22, 19, 59, 43, 95000, tzinfo=datetime.timezone.utc), file_modified_at=datetime.datetime(2022, 12, 22, 19, 59, 45, tzinfo=datetime.timezone.utc), has_metadata=True, is_archived=False, is_favorite=False, is_offline=False, is_trashed=False, local_datetime=datetime.datetime(2022, 12, 22, 20, 59, 43, 95000, tzinfo=datetime.timezone.utc), original_file_name='20221222_205943.jpg', original_path='upload/upload/e7ef5713-9dab-4bd4-b899-715b0ca4379e/43/ab/43abc8a2-3f81-4afc-95ea-c784cf869e4c.jpg', owner_id='e7ef5713-9dab-4bd4-b899-715b0ca4379e', people=[], resized=True, thumbhash='kWkGDAR9CGRpWO11VpJ2cGtVCQ==', updated_at=datetime.datetime(2025, 5, 11, 10, 13, 49, 590877, tzinfo=datetime.timezone.utc), duplicate_id=None, exif_info=ExifInfo(city=None, country=None, date_time_original=None, description=None, exif_image_height=None, exif_image_width=None, exposure_time=None, file_size_in_byte=None, f_number=None, focal_length=None, iso=None, latitude=None, lens_model=None, longitude=None, make=None, model=None, modify_date=None, orientation=None, projection_type=None, rating=None, state=None, time_zone=None), library_id=None, live_photo_video_id=None, original_mime_type='image/jpeg', visibility=None),
    ImmichAsset(asset_id='7d9ecb29-b12a-498d-adf9-c6ac09c761d3', asset_type=<AssetType.IMAGE: 'IMAGE'>, checksum='8fDGDOnOCVkJK8KiNu1bSmytEQ4=', device_asset_id='web-20221222_185242.jpg-1671731574000', device_id='WEB', duration='0:00:00.00000', file_created_at=datetime.datetime(2022, 12, 22, 17, 52, 44, 97000, tzinfo=datetime.timezone.utc), file_modified_at=datetime.datetime(2022, 12, 22, 17, 52, 54, tzinfo=datetime.timezone.utc), has_metadata=True, is_archived=False, is_favorite=False, is_offline=False, is_trashed=False, local_datetime=datetime.datetime(2022, 12, 22, 18, 52, 44, 97000, tzinfo=datetime.timezone.utc), original_file_name='20221222_185242.jpg', original_path='upload/upload/e7ef5713-9dab-4bd4-b899-715b0ca4379e/4f/a7/4fa7280f-16e6-4eb8-938a-89770266a20d.jpg', owner_id='e7ef5713-9dab-4bd4-b899-715b0ca4379e', people=[], resized=True, thumbhash='FUoGNAKHV4f4ZoLXZ1d2cl9H9w==', updated_at=datetime.datetime(2025, 5, 11, 10, 13, 49, 590479, tzinfo=datetime.timezone.utc), duplicate_id=None, exif_info=ExifInfo(city=None, country=None, date_time_original=None, description=None, exif_image_height=None, exif_image_width=None, exposure_time=None, file_size_in_byte=None, f_number=None, focal_length=None, iso=None, latitude=None, lens_model=None, longitude=None, make=None, model=None, modify_date=None, orientation=None, projection_type=None, rating=None, state=None, time_zone=None), library_id=None, live_photo_video_id=None, original_mime_type='image/jpeg', visibility=None),
    ImmichAsset(asset_id='1ed37a88-c720-49b9-bc3d-0e78aab3099a', asset_type=<AssetType.IMAGE: 'IMAGE'>, checksum='RA2WL5H511G+A7f1hV8+jJdfUUQ=', device_asset_id='web-20221115_185642.jpg-1668535006000', device_id='WEB', duration='0:00:00.00000', file_created_at=datetime.datetime(2022, 11, 15, 17, 56, 42, 77000, tzinfo=datetime.timezone.utc), file_modified_at=datetime.datetime(2022, 11, 15, 17, 56, 46, tzinfo=datetime.timezone.utc), has_metadata=True, is_archived=False, is_favorite=False, is_offline=False, is_trashed=False, local_datetime=datetime.datetime(2022, 11, 15, 18, 56, 42, 77000, tzinfo=datetime.timezone.utc), original_file_name='20221115_185642.jpg', original_path='upload/upload/e7ef5713-9dab-4bd4-b899-715b0ca4379e/7c/86/7c8643f0-f91e-4926-8e6a-8e8c4403e4d0.jpg', owner_id='e7ef5713-9dab-4bd4-b899-715b0ca4379e', people=[], resized=True, thumbhash='UBgODIJvdKmYZmZ2Z3cPlfUjaQ==', updated_at=datetime.datetime(2025, 5, 11, 10, 13, 49, 766801, tzinfo=datetime.timezone.utc), duplicate_id=None, exif_info=ExifInfo(city=None, country=None, date_time_original=None, description=None, exif_image_height=None, exif_image_width=None, exposure_time=None, file_size_in_byte=None, f_number=None, focal_length=None, iso=None, latitude=None, lens_model=None, longitude=None, make=None, model=None, modify_date=None, orientation=None, projection_type=None, rating=None, state=None, time_zone=None), library_id=None, live_photo_video_id=None, original_mime_type='image/jpeg', visibility=None),
  ])
# ---
# name: test_get_all_by_person_ids
  list([
    ImmichAsset(asset_id='2e94c203-50aa-4ad2-8e29-56dd74e0eff4', asset_type=<AssetType.IMAGE: 'IMAGE'>, checksum='HJm7TVOP80S+eiYZnAhWyRaB/Yc=', device_asset_id='web-20230131_182037.jpg-1675185639000', device_id='WEB', duration='0:00:00.00000', file_created_at=datetime.datetime(2023, 1, 31, 17, 20, 37, 85000, tzinfo=datetime.timezone.utc), file_modified_at=datetime.datetime(2023, 1, 31, 17, 20, 39, tzinfo=datetime.timezone.utc), has_metadata=True, is_archived=False, is_favorite=False, is_offline=False, is_trashed=False, local_datetime=datetime.datetime(2023, 1, 31, 18, 20, 37, 85000, tzinfo=datetime.timezone.utc), original_file_name='20230131_182037.jpg', original_path='upload/upload/e7ef5713-9dab-4bd4-b899-715b0ca4379e/b4/b8/b4b8ef00-8a6d-4056-91ff-7f86dc66e427.jpg', owner_id='e7ef5713-9dab-4bd4-b899-715b0ca4379e', people=[], resized=True, thumbhash='1igGFALX8mVGdHc5aChJf5nxNg==', updated_at=datetime.datetime(2025, 5, 11, 10, 13, 49, 590401, tzinfo=datetime.timezone.utc), duplicate_id=None, exif_info=ExifInfo(city=None, country=None, date_time_original=None, description=None, exif_image_height=None, exif_image_width=None, exposure_time=None, file_size_in_byte=None, f_number=None, focal_length=None, iso=None, latitude=None, lens_model=None, longitude=None, make=None, model=None, modify_date=None, orientation=None, projection_type=None, rating=None, state=None, time_zone=None), library_id=None, live_photo_video_id=None, original_mime_type='image/jpeg', visibility=None),
    ImmichAsset(asset_id='de104b75-6c7f-46a2-83ce-12a5b1c3a3b4', asset_type=<AssetType.IMAGE: 'IMAGE'>, checksum='aNCzp51jAuT3rXMv1icKLGMTC6M=', device_asset_id='web-20221222_205943.jpg-1671739185000', device_id='WEB', duration='0:00:00.00000', file_created_at=datetime.datetime(2022, 12, 22, 19, 59, 43, 95000, tzinfo=datetime.timezone.utc), file_modified_at=datetime.datetime(2022, 12, 22, 19, 59, 45, tzinfo=datetime.timezone.utc), has_metadata=True, is_archived=False, is_favorite=False, is_offline=False, is_trashed=False, local_datetime=datetime.datetime(2022, 12, 22, 20, 59, 43, 95000, tzinfo=datetime.timezone.utc), original_file_name='20221222_205943.jpg', original_path='upload/upload/e7ef5713-9dab-4bd4-b899-715b0ca4379e/43/ab/43abc8a2-3f81-4afc-95ea-c784cf869e4c.jpg', owner_id='e7ef5713-9dab-4bd4-b899-715b0ca4379e', people=[], resized=True, thumbhash='kWkGDAR9CGRpWO11VpJ2cGtVCQ==', updated_at=datetime.datetime(2025, 5, 11, 10, 13, 49, 590877, tzinfo=datetime.timezone.utc), duplicate_id=None, exif_info=ExifInfo(city=None, country=None, date_time_original=None, description=None, exif_image_height=None, exif_image_width=None, exposure_time=None, file_size_in_byte=None, f_number=None, focal_length=None, iso=None, latitude=None, lens_model=None, longitude=None, make=None, model=None, modify_date=None, orientation=None, projection_type=None, rating=None, state=None, time_zone=None), library_id=None, live_photo_video_id=None, original_mime_type='image/jpeg', visibility=None),
    ImmichAsset(asset_id='7d9ecb29-b12a-498d-adf9-c6ac09c761d3', asset_type=<AssetType.IMAGE: 'IMAGE'>, checksum='8fDGDOnOCVkJK8KiNu1bSmytEQ4=', device_asset_id='web-20221222_185242.jpg-1671731574000', device_id='WEB', duration='0:00:00.00000', file_created_at=datetime.datetime(2022, 12, 22, 17, 52, 44, 97000, tzinfo=datetime.timezone.utc), file_modified_at=datetime.datetime(2022, 12, 22, 17, 52, 54, tzinfo=datetime.timezone.utc), has_metadata=True, is_archived=False, is_favorite=False, is_offline=False, is_trashed=False, local_datetime=datetime.datetime(2022, 12, 22, 18, 52, 44, 97000, tzinfo=datetime.timezone.utc), original_file_name='20221222_185242.jpg', original_path='upload/upload/e7ef5713-9dab-4bd4-b899-715b0ca4379e/4f/a7/4fa7280f-16e6-4eb8-938a-89770266a20d.jpg', owner_id='e7ef5713-9dab-4bd4-b899-715b0ca4379e', people=[], resized=True, thumbhash='FUoGNAKHV4f4ZoLXZ1d2cl9H9w==', updated_at=datetime.datetime(2025, 5, 11, 10, 13, 49, 590479, tzinfo=datetime.timezone.utc), duplicate_id=None, exif_info=ExifInfo(city=None, country=None, date_time_original=None, description=None, exif_image_height=None, exif_image_width=None, exposure_time=None, file_size_in_byte=None, f_number=None, focal_length=None, iso=None, latitude=None, lens_model=None, longitude=None, make=None, model=None, modify_date=None, orientation=None, projection_type=None, rating=None, state=None, time_zone=None), library_id=None, live_photo_video_id=None, original_mime_type='image/jpeg', visibility=None),
    ImmichAsset(asset_id='1ed37a88-c720-49b9-bc3d-0e78aab3099a', asset_type=<AssetType.IMAGE: 'IMAGE'>, checksum='RA2WL5H511G+A7f1hV8+jJdfUUQ=', device_asset_id='web-20221115_185642.jpg-1668535006000', device_id='WEB', duration='0:00:00.00000', file_created_at=datetime.datetime(2022, 11, 15, 17, 56, 42, 77000, tzinfo=datetime.timezone.utc), file_modified_at=datetime.datetime(2022, 11, 15, 17, 56, 46, tzinfo=datetime.timezone.utc), has_metadata=True, is_archived=False, is_favorite=False, is_offline=False, is_trashed=False, local_datetime=datetime.datetime(2022, 11, 15, 18, 56, 42, 77000, tzinfo=datetime.timezone.utc), original_file_name='20221115_185642.jpg', original_path='upload/upload/e7ef5713-9dab-4bd4-b899-715b0ca4379e/7c/86/7c8643f0-f91e-4926-8e6a-8e8c4403e4d0.jpg', owner_id='e7ef5713-9dab-4bd4-b899-715b0ca4379e', people=[], resized=True, thumbhash='UBgODIJvdKmYZmZ2Z3cPlfUjaQ==', updated_at=datetime.datetime(2025, 5, 11, 10, 13, 49, 766801, tzinfo=datetime.timezone.utc), duplicate_id=None, exif_info=ExifInfo(city=None, country=None, date_time_original=None, description=None, exif_image_height=None, exif_image_width=None, exposure_time=None, file_size_in_byte=None, f_number=None, focal_length=None, iso=None, latitude=None, lens_model=None, longitude=None, make=None, model=None, modify_date=None, orientation=None, projection_type=None, rating=None, state=None, time_zone=None), library_id=None, live_photo_video_id=None, original_mime_type='image/jpeg', visibility=None),
  ])
# ---
# name: test_get_all_by_tag_ids
  list([
    ImmichAsset(asset_id='2e94c203-50aa-4ad2-8e29-56dd74e0eff4', asset_type=<AssetType.IMAGE: 'IMAGE'>, checksum='HJm7TVOP80S+eiYZnAhWyRaB/Yc=', device_asset_id='web-20230131_182037.jpg-1675185639000', device_id='WEB', duration='0:00:00.00000', file_created_at=datetime.datetime(2023, 1, 31, 17, 20, 37, 85000, tzinfo=datetime.timezone.utc), file_modified_at=datetime.datetime(2023, 1, 31, 17, 20, 39, tzinfo=datetime.timezone.utc), has_metadata=True, is_archived=False, is_favorite=False, is_offline=False, is_trashed=False, local_datetime=datetime.datetime(2023, 1, 31, 18, 20, 37, 85000, tzinfo=datetime.timezone.utc), original_file_name='20230131_182037.jpg', original_path='upload/upload/e7ef5713-9dab-4bd4-b899-715b0ca4379e/b4/b8/b4b8ef00-8a6d-4056-91ff-7f86dc66e427.jpg', owner_id='e7ef5713-9dab-4bd4-b899-715b0ca4379e', people=[], resized=True, thumbhash='1igGFALX8mVGdHc5aChJf5nxNg==', updated_at=datetime.datetime(2025, 5, 11, 10, 13, 49, 590401, tzinfo=datetime.timezone.utc), duplicate_id=None, exif_info=ExifInfo(city=None, country=None, date_time_original=None, description=None, exif_image_height=None, exif_image_width=None, exposure_time=None, file_size_in_byte=None, f_number=None, focal_length=None, iso=None, latitude=None, lens_model=None, longitude=None, make=None, model=None, modify_date=None, orientation=None, projection_type=None, rating=None, state=None, time_zone=None), library_id=None, live_photo_video_id=None, original_mime_type='image/jpeg', visibility=None),
    ImmichAsset(asset_id='de104b75-6c7f-46a2-83ce-12a5b1c3a3b4', asset_type=<AssetType.IMAGE: 'IMAGE'>, checksum='aNCzp51jAuT3rXMv1icKLGMTC6M=', device_asset_id='web-20221222_205943.jpg-1671739185000', device_id='WEB', duration='0:00:00.00000', file_created_at=datetime.datetime(2022, 12, 22, 19, 59, 43, 95000, tzinfo=datetime.timezone.utc), file_modified_at=datetime.datetime(2022, 12, 22, 19, 59, 45, tzinfo=datetime.timezone.utc), has_metadata=True, is_archived=False, is_favorite=False, is_offline=False, is_trashed=False, local_datetime=datetime.datetime(2022, 12, 22, 20, 59, 43, 95000, tzinfo=datetime.timezone.utc), original_file_name='20221222_205943.jpg', original_path='upload/upload/e7ef5713-9dab-4bd4-b899-715b0ca4379e/43/ab/43abc8a2-3f81-4afc-95ea-c784cf869e4c.jpg', owner_id='e7ef5713-9dab-4bd4-b899-715b0ca4379e', people=[], resized=True, thumbhash='kWkGDAR9CGRpWO11VpJ2cGtVCQ==', updated_at=datetime.datetime(2025, 5, 11, 10, 13, 49, 590877, tzinfo=datetime.timezone.utc), duplicate_id=None, exif_info=ExifInfo(city=None, country=None, date_time_original=None, description=None, exif_image_height=None, exif_image_width=None, exposure_time=None, file_size_in_byte=None, f_number=None, focal_length=None, iso=None, latitude=None, lens_model=None, longitude=None, make=None, model=None, modify_date=None, orientation=None, projection_type=None, rating=None, state=None, time_zone=None), library_id=None, live_photo_video_id=None, original_mime_type='image/jpeg', visibility=None),
    ImmichAsset(asset_id='7d9ecb29-b12a-498d-adf9-c6ac09c761d3', asset_type=<AssetType.IMAGE: 'IMAGE'>, checksum='8fDGDOnOCVkJK8KiNu1bSmytEQ4=', device_asset_id='web-20221222_185242.jpg-1671731574000', device_id='WEB', duration='0:00:00.00000', file_created_at=datetime.datetime(2022, 12, 22, 17, 52, 44, 97000, tzinfo=datetime.timezone.utc), file_modified_at=datetime.datetime(2022, 12, 22, 17, 52, 54, tzinfo=datetime.timezone.utc), has_metadata=True, is_archived=False, is_favorite=False, is_offline=False, is_trashed=False, local_datetime=datetime.datetime(2022, 12, 22, 18, 52, 44, 97000, tzinfo=datetime.timezone.utc), original_file_name='20221222_185242.jpg', original_path='upload/upload/e7ef5713-9dab-4bd4-b899-715b0ca4379e/4f/a7/4fa7280f-16e6-4eb8-938a-89770266a20d.jpg', owner_id='e7ef5713-9dab-4bd4-b899-715b0ca4379e', people=[], resized=True, thumbhash='FUoGNAKHV4f4ZoLXZ1d2cl9H9w==', updated_at=datetime.datetime(2025, 5, 11, 10, 13, 49, 590479, tzinfo=datetime.timezone.utc), duplicate_id=None, exif_info=ExifInfo(city=None, country=None, date_time_original=None, description=None, exif_image_height=None, exif_image_width=None, exposure_time=None, file_size_in_byte=None, f_number=None, focal_length=None, iso=None, latitude=None, lens_model=None, longitude=None, make=None, model=None, modify_date=None, orientation=None, projection_type=None, rating=None, state=None, time_zone=None), library_id=None, live_photo_video_id=None, original_mime_type='image/jpeg', visibility=None),
    ImmichAsset(asset_id='1ed37a88-c720-49b9-bc3d-0e78aab3099a', asset_type=<AssetType.IMAGE: 'IMAGE'>, checksum='RA2WL5H511G+A7f1hV8+jJdfUUQ=', device_asset_id='web-20221115_185642.jpg-1668535006000', device_id='WEB', duration='0:00:00.00000', file_created_at=datetime.datetime(2022, 11, 15, 17, 56, 42, 77000, tzinfo=datetime.timezone.utc), file_modified_at=datetime.datetime(2022, 11, 15, 17, 56, 46, tzinfo=datetime.timezone.utc), has_metadata=True, is_archived=False, is_favorite=False, is_offline=False, is_trashed=False, local_datetime=datetime.datetime(2022, 11, 15, 18, 56, 42, 77000, tzinfo=datetime.timezone.utc), original_file_name='20221115_185642.jpg', original_path='upload/upload/e7ef5713-9dab-4bd4-b899-715b0ca4379e/7c/86/7c8643f0-f91e-4926-8e6a-8e8c4403e4d0.jpg', owner_id='e7ef5713-9dab-4bd4-b899-715b0ca4379e', people=[], resized=True, thumbhash='UBgODIJvdKmYZmZ2Z3cPlfUjaQ==', updated_at=datetime.datetime(2025, 5, 11, 10, 13, 49, 766801, tzinfo=datetime.timezone.utc), duplicate_id=None, exif_info=ExifInfo(city=None, country=None, date_time_original=None, description=None, exif_image_height=None, exif_image_width=None, exposure_time=None, file_size_in_byte=None, f_number=None, focal_length=None, iso=None, latitude=None, lens_model=None, longitude=None, make=None, model=None, modify_date=None, orientation=None, projection_type=None, rating=None, state=None, time_zone=None), library_id=None, live_photo_video_id=None, original_mime_type='image/jpeg', visibility=None),
  ])
# ---
